packages/core/src/browser/tree/tree-model.ts

```diff
@@ -5,112 +5,174 @@
+     */
+    selectPrevNode(keepSelection?: boolean): void;
+
+    /**
+     * Selects the next node relatively to the currently selected one. This method takes the expansion state of the tree into consideration.
+     */
+    selectNextNode(keepSelection?: boolean): void;
+
+    /**
+     * Selects the tree node argument.
+     *
+     *  - If the `keepSelection` argument is `false` or `undefined`, then it has the same effect as invoking `setSelection([node])`. The previous selection
+     * state will be overridden with given single tree node.
+     *  - If `keepSelection` is `true`, then the previous selection state will be kept and the `node` argument will be merged into the previous selection state.
+     *  - If `keepSelection` is `true` and the `node` was already selected, the selection content will not change but the order of the selected items. The
+     * `node` argument will be the most recently selected item.
+     */
+    selectNode(node: Readonly<SelectableTreeNode>, keepSelection?: boolean): void;
+
+    /**
+     * Unselects the given tree node argument. Has no effect, if the argument is not a valid tree node, or was not yet selected.
+     */
+    unselectNode(node: Readonly<SelectableTreeNode>): void;
+
+    /**
+     * Toggles the selection state of the `node` argument. If the node was not yet selected, it will be. Toggling the state of the tree node
+     * will keep any previous selection state. If the `node` was selected, `unselects` the node.
+     */
+    toggleSelection(node: Readonly<SelectableTreeNode>): void;
+
+    /**
+     * Returns with the nodes between (inclusive) the `toNode` and the `fromNode`. Returns with an empty array if either the `toNode` or the `fromNode` is invalid.
+     */
+    calculateRange(toNode: Readonly<SelectableTreeNode>, fromNode: Readonly<SelectableTreeNode>): ReadonlyArray<Readonly<SelectableTreeNode>>;
+
+    /**
+     * Calculates and sets the selection range between the `fromNode` and the `toNode`.
+     *  - Has not effect if either the `fromNode` or the `toNode` does not exist or invalid.
+     *  - If `keepSelection` is `true`, then the range of selection will be merged into the previous selection state.
+     *  - If `keepSelection` is `false` or `undefined`, then the previous state will be overridden with the selection range.
+     *  - If `fromNode` is `undefined`, then the most recently selected tree node will be used instead.
+     */
+    selectRange(toNode: Readonly<SelectableTreeNode>, fromNode?: Readonly<SelectableTreeNode>, keepSelection?: boolean): void;
+
+    /**
+     * Counterpart of the `selectRange` method. Calculates and removes the selection of the nodes which contained in the range.
+     */
+    unselectRange(toNode: Readonly<SelectableTreeNode>, fromNode?: Readonly<SelectableTreeNode>): void;
+
 }
 
 @injectable()
 export class TreeServices {
```

Comment:
since you are refactoring, please get rid of it as well and replyc with property injections + `postConstruct`

Reply:
Great idea!